src/nvim/testdir/test_functions.vim

```diff
@@ -869,8 +869,8 @@ func Test_Executable()
     call assert_equal(1, executable('notepad'))
     call assert_equal(1, executable('notepad.exe'))
     call assert_equal(0, executable('notepad.exe.exe'))
-    call assert_equal(1, executable('shell32.dll'))
-    call assert_equal(1, executable('win.ini'))
+    call assert_equal(0, executable('shell32.dll'))
+    call assert_equal(0, executable('win.ini'))
```

Comment:
What does `executable` mean on cmd.exe and sh?
On cmd.exe, running `win.ini` opens notepad on Windows so `executable` should return 1.
On sh, it does not run (because it checks the first line of the file) so it should return 0.

Reply:
`win.ini` is not executable(executable command or executable program), it is a file associated with executable.

@janlazo Do you really think that you would like to search for `foo.txt` or `bar.jpg` using `executable()` from `$PATH`?

As for the UNIX like shell, it is only a current implementation due to the trade-off between implementation cost and usefulness obtained by it.

Reply:
If the user associates perl with `.pl` file and adds `.pl` to `PATHEXT`, the user thinks that `.pl` file should be treated as an executable command or executable program. Therefore, `executable()` must also search for `.pl` files.

Reply:
> Do you really think that you would like to search for foo.txt or bar.jpg using executable() from $PATH?

Yes. If I set `foo.txt` file permissions with Read+Execute and added its directory to `$PATH`, then I expect `executable` to find it and check if it is associated with `.exe` file because that is `executable` enough for me.

I asked what `executable` means now because this PR seems to be for Vim compatibility.

Reply:
I add `.dll` to `PATHEXT`. Is `foo.dll` executable now? cmd.exe cannot run that (by default).